ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java

```diff
@@ -805,13 +808,19 @@ private class CustomGlobalLayoutListener implements ViewTreeObserver.OnGlobalLay
 
     private int mKeyboardHeight = 0;
     private int mDeviceRotation = 0;
+    private boolean mIsTVKeyboardOpen = false;
 
     /* package */ CustomGlobalLayoutListener() {
       DisplayMetricsHolder.initDisplayMetricsIfNotInitialized(getContext().getApplicationContext());
       mVisibleViewArea = new Rect();
       mMinKeyboardHeightDetected = (int) PixelUtil.toPixelFromDIP(60);
     }
 
+    private boolean isTVDevice() {
+      UiModeManager uiModeManager = (UiModeManager) getContext().getSystemService(Context.UI_MODE_SERVICE);
```

Comment:
From a performance point of view we should not call `getSystemService` from a layout method.
Either lazy initialise a field or in in view constructor. I'd recommend constructor as it's easier.